tests/test_util_descriptor.py

```diff
@@ -0,0 +1,291 @@
+        )
+    assert str(excinfo.value) == "Checksum is empty."
+
+    # Checksum too long
+    with pytest.raises(SpecterError) as excinfo:
+        Descriptor.parse(
+            "sh(multi(2,[00000000/111'/222]xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjg09x5tq"
+        )
+    assert (
+        str(excinfo.value)
+        == "Checksum tjg09x5tq doesn't have the correct length. Should be 8 characters not 9."
+    )
+
+    # Checksum too short
+    with pytest.raises(SpecterError) as excinfo:
+        Descriptor.parse(
+            "sh(multi(2,[00000000/111'/222]xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjg09x5"
+        )
+    assert (
+        str(excinfo.value)
+        == "Checksum tjg09x5 doesn't have the correct length. Should be 8 characters not 7."
+    )
+
+    # Error in checksum
+    with pytest.raises(SpecterError) as excinfo:
+        Descriptor.parse(
+            "sh(multi(2,[00000000/111'/222]xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjq09x4t"
+        )
+    assert str(excinfo.value) == "tjq09x4t is the wrong checkum should be tjg09x5t."
+
+
+### Tests of additional descriptor functionality in Specter
+
+
+def test_derive():
+    # Using ghost x 11 + machine (1)
+    # abandon x 11 + about (2)
+    # and zoo x 11 + wrong (3)
+    # /48'/1'/0'/1'
+
+    desc = Descriptor.parse(
+        "sh(wsh(sortedmulti(2,tpubDDzWqfZ5TH4819JtJT1MaJGh2FYnbn2KGoqkznXRFdNZAuKLD2CsYtQiV5rEVCUezzz9GaRkeHct5NSxVEG9KWUaRoeEtcafVHr2SVE5DRN/*,tpubDFH9dgzveyD8yHQb8VrpG8FYAuwcLMHMje2CCcbBo1FpaGzYVtJeYYxcYgRqSTta5utUFts8nPPHs9C2bqoxrey5jia6Dwf9mpwrPq7YvcJ/*,tpubDFPtPArj4GzBBcuqDySkeQbKx4r6HwRgcPbbAjbjB5cxYRzJT6iFtiqzce4qQ9XFWZ83DZJ43WCJJsotdG75p7pw4SgUHZ2nkG4YxLQ414i/2)))",
+        True,
+    )
+
+    assert (
+        desc.derive(1).serialize()
+        == "sh(wsh(sortedmulti(2,03407a711574ae73aa5824f5a66bf4f9a9f49dd274407eb4c27d996019cf4a6552,0376166abb71efb6c9a497a64c1b24c484c29b8a4219a526737e1f370768b1bbe8,03a1427c178f0b1cd679464c4c90444bcfd775d5edb89cf33828170e8be008d921)))#j54e69dq"
+    )
+    # Pubkeys sorted
+    # Verified with iancoleman
+
+    assert desc.address(5) == "2NGamwat67EUkABbkY1HLYjQW3oumUcThnV"
+    # Verified with bitcoin-cli deriveaddresses
+
+    assert desc.address(5, "main") == "388fc825v9R6Ev8BKodXQMFumQRe7C8SZ5"
```

Comment:
I got a different address with bitcoin-cli, but can't rule out a mistake with the xpubs. Would be good, if someone else could check it.

Reply:
Strange, running the same code returns `3R2Zsqx4VmyPxPyCrsfTvnREqThbhjFG4N` for me.

Reply:
Fixed as we discussed, could now produce the correct address that bitcoin-cli gives, see latest commit.